pack/map.h

```diff
@@ -213,7 +217,21 @@ int Map<T>::size() const
 template <typename T>
 const T& Map<T>::operator[](const std::string& key) const
 {
-    auto found = std::find_if(m_value.begin(), m_value.end(), [&](const auto& pair) {
+    const auto found = std::find_if(m_value.begin(), m_value.end(), [&](const auto& pair) {
+        return pair.first == key;
+    });
```

Comment:
You will search by key? Use https://en.cppreference.com/w/cpp/container/map/find or https://en.cppreference.com/w/cpp/container/map/count